test/jruby/test_zlib.rb

```diff
@@ -517,6 +517,24 @@ def test_error_input
     assert_equal("not in gzip format", e.message)
     assert_equal("foobarzothoge", e.input)
   end
+
+  def test_gzip_reader_ungetc
```

Comment:
Makes sense. There are placeholders for these methods in ruby/spec, so I'll add them there.

Should I submit them to ruby/spec first and then pull them in once merged upstream?